extension/js/common/core/common.ts

```diff
@@ -256,11 +256,16 @@ export class Str {
     return date.substr(0, 10).replace(/"/g, '&quot;').replace(/'/g, '&#39;').replace(/</g, '&lt;');
   };
 
-  public static fromDate = (date: Date) => {
-    return date
-      .toISOString()
-      .replace(/T/, ' ')
-      .replace(/:[^:]+$/, '');
+  public static fromDate = (date: Date, ishour12Format = false) => {
+    const formatOptions: Intl.DateTimeFormatOptions = {
+      year: 'numeric',
+      month: '2-digit',
+      day: '2-digit',
+      hour: '2-digit',
+      minute: '2-digit',
+      hour12: ishour12Format,
+    };
+    return date.toLocaleString('en-CA', formatOptions).replace(/,/g, ''); // en-CA produces yyyy-dd-mm, hh:mm format
```

Comment:
What about using current user locale like `date.toLocaleString([], formatOptions)`, as for US date locale day precedes month, so it's `dd-mm` instead of `mm-dd`. By using user's locale we'll have date formatting usual for them

Reply:
The usage of "en-CA" produces yyyy-dd-mm format with a dash separator itself compared to using `[]` or `undefined` value passed into the toLocaleString. its the most concise option in my opinion compared to using `[]` then replacing '/' with '-' not to mention that it will result to dd/mm/yyyy format instead of the preferred yyyy-dd-mm... But please let me know if you prefer the "dd/mm/yyyy" format over the original "yyyy-dd-mm" and will change it immediately.

Reply:
> its the most concise option in my opinion compared to using [] then replacing '/' with '-' not to mention that it will result to dd/mm/yyyy format instead of the preferred yyyy-dd-mm

`date.toLocaleString([], formatOptions)` will produce date format specified by user browser locale (so it'll be usual for user), and there is no need to replace `/` with `-`. For example, user in Germany will get `27.04.2024` date string, US user - `04/27/2024` and so on

Reply:
Hello @sosnovsky, I've encountered an issue. I noticed that if I passed `[]` as the defined Locale format for "toLocaleString", the test results would differ for each of us. For example, if I make changes and use the correct static time for myself, such as in the "forward - pgp/mime signed-only" test where it passed for me, the time returned in the CI test environment would be different. I believe we need to establish a standardized date format to follow, considering the diverse formats dates can offer.

![image](https://github.com/FlowCrypt/flowcrypt-browser/assets/46025304/cc028a90-269a-41ee-98c6-255cb836809f)

On CI test:
![image](https://github.com/FlowCrypt/flowcrypt-browser/assets/46025304/d5af0f73-e731-49dc-927a-64e569c33108)

Reply:
Can we then use `toLocaleString` only for the print view and for `fromDate` we should just fix incorrect timezone by adding current user timezone offset like this:
```js
var timeZoneOffset = (new Date()).getTimezoneOffset() * 60000; //offset in milliseconds
var localTime = new Date(date.getTime() + timeZoneOffset);
```

Reply:
Hello @Roma I got the same output just like the ".toISOString"... I think we have just concluded that the toISOString produces inaccurate output when its not as I check the original email with this code snippet and the code we had before (.toISOString) and produces the same time... So I think its just looks wrong since its not displayed in user locale format in print...

Could we let use the .toLocaleString only to print as we'll get different date and time outputs if we use toLocaleString across tests.

Reply:
> Could we let use the .toLocaleString only to print as we'll get different date and time outputs if we use toLocaleString across tests.

Sounds good, let's use it only for print view